cmd/collectiondir/serve.go

```
@@ -705,6 +704,9 @@ func (cs *collectionServer) ingestReceiver() {
 			} else {
 				errcount = 0
 			}
+		case <-cs.shutdown:
```

I'm not sure I have a strong preference for either of our approaches to solve this ([here's](https://github.com/bluesky-social/indigo/pull/1029/commits/8c00259a165833ad0e8e8870691b5f55e3725ea8) what I did). Way too many goroutines flying around for managing the server listeners and there's just a lot of mental load needed to read this file from top to bottom.

Happy to take either solution, but I think if we do take mine, we should change `StartApiServer` and `StartMetricsServer` to fatally shut down on any error rather than just return errors which will now be silently eaten. I like that mine simplifies the # of goroutines needed though (made that change [here](https://github.com/bluesky-social/indigo/pull/1029/commits/444a6aa911744316ac491b404a3a793311bfddfe)).

---

I'm in favor of your solution, and let's come back to it after this PR is merged.